lib/fabrik/VirtualHostAgent.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const Agent = require('./Agent');
+
+class VirtualHostAgent extends Agent {
+
+  constructor(settings) {
+    super({
+      json: true
+    });
+    this.settings = settings;
+  }
+
+  provision(ips, guid) {
```

Comment:
for the sake of consistence across clasess can this be renamed to `createVirtualHost` or `crerateVhost` - the same operation is called differently along the flow from controller to instance -> manager -> agent.